Source/Public/Test-VSTeamYamlPipeline.ps1

```diff
@@ -0,0 +1,51 @@
+function Test-VSTeamYamlPipeline {   
+   [CmdletBinding(DefaultParameterSetName = 'WithFilePath')]
+   param(
+      [Parameter(ParameterSetName = 'WithFilePath', Mandatory = $true, ValueFromPipelineByPropertyName = $true, Position = 1)]
+      [Parameter(ParameterSetName = 'WithYamlOverride', Mandatory = $true, ValueFromPipelineByPropertyName = $true, Position = 1)]
+      [Int32] $PipelineId,
+      [Parameter(ParameterSetName = 'WithFilePath', Mandatory = $false)]
+      [string] $FilePath,
+      [Parameter(ParameterSetName = 'WithYamlOverride', Mandatory = $false)]
+      [string] $YamlOverride
+   )
+   DynamicParam {
+      _buildProjectNameDynamicParam
+   }
+
+   process {
+      # Bind the parameter to a friendly variable
+      $ProjectName = $PSBoundParameters["ProjectName"]
+
+      $body = @{
+         PreviewRun = $true      
+      }
+
+      if ($FilePath) {
+         $body.YamlOverride = [string](Get-Content -raw $FilePath)
+      }
+      elseif ($YamlOverride) {
+         $body.YamlOverride = $YamlOverride         
+      }
+
+      try {
+         # Call the REST API
+         $resp = _callAPI -ProjectName $ProjectName -Area 'pipelines' -Resource "$PipelineId" -id "runs" `
+            -Method Post -ContentType 'application/json' -Body ($body | ConvertTo-Json) `
```

Comment:
; charset=utf-8
:)

Reply:
we need to make that maybe part of the build pipeline probably of the API call to not allow non utf json calls.